src/Http/Place/LegacyPlaceRequestBodyParser.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Place;
+
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use Psr\Http\Message\ServerRequestInterface;
+use stdClass;
+
+final class LegacyPlaceRequestBodyParser implements RequestBodyParser
+{
+    public function parse(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $data = $request->getParsedBody();
+
+        if (!($data instanceof stdClass)) {
+            return $request;
+        }
+        $data = clone $data;
+
+        $mainLanguage = $data->mainLanguage ?? null;
+
+        if ($mainLanguage && isset($data->name) && is_string($data->name)) {
+            $data->name = (object) [
+                $data->mainLanguage => $data->name,
+            ];
+        }
+
+        if (isset($data->type) && $data->type instanceof stdClass) {
+            $data->terms = [
+                (object) [
+                    'id' => $data->type->id,
+                    'label' => $data->type->id,
+                    'domain' => $data->type->id,
+                ],
+            ];
+        }
+
+        if ($mainLanguage && isset($data->address->streetAddress)) {
+            $data->address->{$mainLanguage} = $data->address->{$mainLanguage} ?? (object) [];
+            $data->address->{$mainLanguage}->streetAddress = $data->address->streetAddress;
+            unset($data->address->streetAddress);
+        }
+        if ($mainLanguage && isset($data->address->postalCode)) {
+            $data->address->{$mainLanguage} = $data->address->{$mainLanguage} ?? (object) [];
+            $data->address->{$mainLanguage}->postalCode = $data->address->postalCode;
+            unset($data->address->postalCode);
+        }
+        if ($mainLanguage && isset($data->address->addressLocality)) {
+            $data->address->{$mainLanguage} = $data->address->{$mainLanguage} ?? (object) [];
+            $data->address->{$mainLanguage}->addressLocality = $data->address->addressLocality;
+            unset($data->address->addressLocality);
+        }
+        if ($mainLanguage && isset($data->address->addressCountry)) {
+            $data->address->{$mainLanguage} = $data->address->{$mainLanguage} ?? (object) [];
+            $data->address->{$mainLanguage}->addressCountry = $data->address->addressCountry;
+            unset($data->address->addressCountry);
+        }
+
+        if (isset($data->calendar) && $data->calendar instanceof stdClass) {
+            $data->calendarType = $data->calendar->calendarType;
```

Comment:
`calendarType` is not guaranteed to be in the request body

Reply:
Should be fixed.